src/parquet/column/levels.h

```diff
@@ -82,5 +83,32 @@ class PARQUET_EXPORT LevelDecoder {
   std::unique_ptr<BitReader> bit_packed_decoder_;
 };
 
+// ----------------------------------------------------------------------
+// Definition and Repetition level vectors
+
+typedef std::vector<int16_t> ValueLevels;
+typedef std::shared_ptr<std::vector<int16_t>> ValueLevelsPtr;
```

Comment:
My thinking was that the lower level API could benefit not messing around with all the int16_t* (even though I haven't implemented it in this PR).
Didn't take into account the pool allocators though. I'll use a PoolBuffer in `arrow/reader.cc` instead.